src/test/rgw/rgw_multi/tests.py

```diff
@@ -586,6 +586,42 @@ def test_bucket_create():
     for zone in zonegroup_conns.zones:
         assert check_all_buckets_exist(zone, buckets)
 
+def test_bucket_create_with_tenant():
+    zonegroup = realm.master_zonegroup()
+    zonegroup_conns = ZonegroupConns(zonegroup)
+    primary = zonegroup_conns.rw_zones[0]
+    secondary = zonegroup_conns.rw_zones[1]
+
+    access_key = 'abcd'
+    secret_key = 'efgh'
+    tenant = 'testx'
+    uid = 'test'
+
+    cmd = ['user', 'create', '--tenant', tenant, '--uid', uid, '--access-key', access_key, '--secret-key', secret_key, '--display-name', 'tenanted-user']
+    primary.zone.cluster.admin(cmd)
+
+    zonegroup_meta_checkpoint(zonegroup)
+
+    secondary_conn = boto.s3.connection.S3Connection(aws_access_key_id=access_key,
+                                aws_secret_access_key=secret_key,
+                                is_secure=False,
+                                port=secondary.zone.gateways[0].port,
+                                host=secondary.zone.gateways[0].host,
+                                calling_format='boto.s3.connection.OrdinaryCallingFormat')
+
+    secondary_conn.create_bucket('tenanted-bucket')
+    bucket_name = 'testx:tenanted-bucket'
```

Comment:
this kind of s3 cross-tenant access is now deprecated (see https://github.com/ceph/ceph/pull/60848). can we just verify that `tenanted-bucket` exists for `secondary_conn` but not `primary_conn`?

Reply:
ok. since we cannot use s3 get_bucket(), I have updated the test to upload an object on secondary and use zone_bucket_checkpoint() on primary to validate the bucket.

Reply:
i was just suggesting `get_bucket("tenanted-bucket")` to test whether the name exists in each user's namespace

Reply:
done! I had misunderstood your comment sorry. and `primary_conn` is the same tenant user connection but on master zone. bad naming choice. corrected it now :)

Reply:
changes look great 👍